internal/node/client_pool.go

```diff
@@ -2,12 +2,12 @@ package client
 
 import (
 	"github.com/pkg/errors"
-	"github.com/threefoldtech/terraform-provider-grid/pkg/subi"
+	"github.com/threefoldtech/substrate-client"
 	"github.com/threefoldtech/zos/pkg/rmb"
 )
 
 type NodeClientCollection interface {
-	GetNodeClient(sub subi.SubstrateExt, nodeID uint32) (*NodeClient, error)
+	GetNodeClient(sub *substrate.Substrate, nodeID uint32) (*NodeClient, error)
```

Comment:
should be named `NodeClientGetter` instead of NodeClientCollection, as this has no indication on the interface behavior